state/keys.go

```diff
@@ -4,10 +4,9 @@
 package state
 
 const (
-	Read Permissions = 1 << iota
-	// TODO: Handle Allocate permission
-	Allocate
-	Write
+	Read     Permissions = 1
+	Allocate             = 2 | Read
+	Write                = 4 | Read
```

Comment:
nit: I would recommend keeping the bit shifts here instead of computing yourself:
1<<1
1<<2 | Read
1<<3 | Read